test/expr/closure/closures_swift6.swift

```diff
@@ -561,6 +647,12 @@ class TestGithubIssue69911 {
           }
         }
       }
+
+      doVoidStuff { [weak self] in
+        doVoidStuff { [self] in // expected-error {{value of optional type 'TestGithubIssue69911?' must be unwrapped to a value of type 'TestGithubIssue69911'}} expected-note{{coalesce using '??' to provide a default when the optional value contains 'nil'}} expected-note{{force-unwrap using '!' to abort execution if the optional value contains 'nil'}}
```

Comment:
Shouldn't this diagnostic be on the next line? This also looks like the same test case that's covered above in `testClosuresInsideWeakSelfNotUnwrapped`?

Reply:
I agree this isn't what I expect to see, but I checked and this issue isn't introduced by this PR. I get the exact same error in the same location when running this code on the unmodified `swift-DEVELOPMENT-SNAPSHOT-2025-01-31-` tag. As long as this is rejected as we expect then I think it's ok in the context of this PR.

Reply:
Hmm I think this may have changed recently, when rebasing this change on main, I get the expected diagnostic on the implicit use of `self` rather than the capture